### PR TITLE
feat: Path-based component resolution for all commands

### PR DESCRIPTION
**Pull request by osterman**
## Summary

Enables developers to use filesystem paths (e.g., `.`, `./vpc`) instead of component names with all component-related commands. Paths are automatically resolved to component names based on stack configuration, eliminating the need to memorize component names.

## Changes

- **Core Implementation**: Path extraction, normalization, and stack validation utilities with symlink resolution
- **Commands Enhanced**: Terraform, Helmfile, Packer, `describe component`, and `validate component` commands
- **Tab Completion**: Enhanced completion with path support and graceful degradation when resolution fails
- **Testing**: 17 integration test scenarios + 6 unit tests covering all use cases
- **Documentation**: Full documentation with examples for all command types + blog post

## Example Usage

```bash
# Navigate to component
cd components/terraform/vpc

# Use . to reference current directory
atmos terraform plan . --stack dev
atmos describe component . --stack dev
atmos validate component . --stack dev
```

## Features

- Auto-detection of component type from filesystem path
- Stack validation to ensure component exists in specified stack
- Clear error messages when paths cannot be resolved
- Backward compatible - existing component name syntax still works

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Path-based component resolution (., ./, ../, absolute) with auto-type detection, stack validation, interactive disambiguation, and directory-aware tab completion.

* **Bug Fixes**
  * Commands propagate errors instead of exiting; config now exposes absolute base-paths and clearer, actionable path/stack errors.

* **Documentation**
  * PRD, CLI usage pages, and blog post updated with path-based examples and guidance.

* **Tests**
  * Extensive new/updated unit, integration, and golden snapshot tests covering path resolution, completions, and edge cases.

* **Other**
  * Telemetry notice added; CLI help lists a new terraform subcommand ("provision"); legacy CLI config file removed.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by osterman** [comment]
@coderabbitai review

File: pkg/utils/component_path_utils.go [hunk 1]
```diff
@@ -124,8 +131,9 @@ func cleanDuplicatedPath(path string) string {
 		startIdx = 1
 	}
 
-	// Only look for duplications of sequences that are at least 3 parts long
-	minLength := 3
+	// Only look for duplications of sequences that are at least 2 parts long.
+	// This catches patterns like /tests/fixtures/tests/fixtures (2-part duplication).
+	minLength := 2
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/utils/component_path_utils.go:155]: ✅ **Resolved in commit 17773a40b**

Added comprehensive unit tests for `CleanDuplicatedPath` with `minLength=2` in `pkg/utils/component_path_utils_test.go`.

Test coverage includes:
- 2-part duplication patterns (e.g., `/a/b/a/b/c` → `/a/b/c`)
- Recursive/multiple consecutive duplications (e.g., `/a/b/a/b/a/b/c` → `/a/b/c`)
- Edge cases where 2-part sequences legitimately repeat without being duplications
- Cross-platform path handling (Windows and Unix)
- Real-world symlink scenario (tests/fixtures duplication)

All 22 test cases pass successfully.

File: pkg/utils/component_reverse_path_utils.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+				dirDesc, absPath, capitalizeFirst(dirDesc), dirPath).
+			WithHint("Components are located in component directories (terraform, helmfile, packer)\nChange to a component directory and use `.` or provide a path within a component directory").
+			WithContext("path", absPath).
+			WithContext(contextKey, dirPath).
+			WithExitCode(2).
+			Err()
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/utils/component_reverse_path_utils.go]: ✅ **Resolved in commit 17773a40b**

Refactored `validatePathIsNotConfigDirectory` to prefer pre-resolved absolute path fields from config:

- Added `getStacksBasePath()` helper that uses `atmosConfig.StacksBaseAbsolutePath` when available, falling back to `resolveBasePath(atmosConfig.Stacks.BasePath)` only if needed
- Added `getWorkflowsBasePath()` helper for consistency and future-proofing (includes TODO for when `WorkflowsBaseAbsolutePath` is added to schema)

This ensures path resolution is consistent with the rest of the config processing and avoids CWD-based drift when atmos is invoked from different working directories.

**Comment by aknysh** [comment]
@coderabbitai full review please

**Comment by aknysh** [comment]
@coderabbitai please validate that your recent comments have been correctly addressed

**Comment by aknysh** [comment]
@coderabbitai full review please
